### PR TITLE
Refactor updates settings

### PR DESCRIPTION
**Pull request by yourcontact**
### Description

- I renamed the `Disable updates` toggle to `Enable automatic updates`.
- I reordered `Updates` settings: the toggle `Enable automatic updates` is now first in the list (from third.)
- I added a condition for the 2 other settings: `Include beta versions` and the `Update` button to show if `Enable automatic updates` is toggled on.

### Motivation and Context

- The name of the toggle to enable updates was confusing.
- The order was illogical. The toggle to enable updates was last.
- `Include beta versions` toggle and the `Update` button showed even if the user did not enable updates.

### How Has This Been Tested?

I tested the changes in Electron.

### Screenshots (if appropriate):

Before the changes were made:

Screen Shot 2020-03-28 at 10 37 05 AM

After the changes were made with updates disabled:

Screen Shot 2020-03-28 at 10 38 31 AM

After the changes were made with updates enabled:

Screen Shot 2020-03-28 at 10 40 45 AM

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactoring

### Checklist:

- [ ] My code follows the code style of this project (run `$ yarn lint`).

@kytwb we have not talked about "run `$ yarn lint`." Is that something I should do for this change?

### CODE REVIEW

**Comment by vantezzen** [comment]
Great contribution!
It looks like `automaticUpdates` currently gets used exactly like `noUpdates` has been used before - meaning updates would be enabled only when `automaticUpdates` is disabled - you will need to change "src/electron/ipc-api/autoUpdate.js" to fix that.

Updates should be also be enabled by default (it looks like they are currently disabled by default? At least your config says `automaticUpdates: false`).

**Comment by yourcontact** [comment]
Thank you @vantezzen 

> It looks like automaticUpdates currently gets used exactly like noUpdates has been used before - meaning updates would be enabled only when automaticUpdates is disabled - you will need to change "src/electron/ipc-api/autoUpdate.js" to fix that.

I tried a fix https://github.com/getferdi/ferdi/pull/506/commits/d0486015c001ec9eca7927ba420eb40bdde69d1e. I do not know how to test it though.

> Updates should be also be enabled by default (it looks like they are currently disabled by default? At least your config says automaticUpdates: false).

Yes, automatic updates were disabled by default. I changed the config to `automaticUpdates: true` https://github.com/getferdi/ferdi/pull/506/commits/908bcb79d3efeab807f275945c3b64b5b99424a8

**Comment by vantezzen** [comment]
Updates are still enabled when the option is disabled. Your code currently says:
```JavaScript
if (enableUpdates) {
    => disable Updates
}
```
which is the wrong way around. Also, it would be good to remove the second "disableUpdates" variable and use "!enableUpdates" for the check instead.

**Comment by yourcontact** [comment]
Thank you @vantezzen. I will fix [this](https://github.com/getferdi/ferdi/pull/506#issuecomment-605429669) with @kytwb tomorrow.

**Comment by kytwb** [comment]
@all-contributors please add yourcontact for code.

File: src/electron/ipc-api/autoUpdate.js [hunk 1]
```diff
@@ -6,14 +6,13 @@ const debug = require('debug')('Ferdi:ipcApi:autoUpdate');
...
-  if (enableUpdate) {
+  if (!enableUpdate) {
     autoUpdater.autoInstallOnAppQuit = false;
     autoUpdater.autoDownload = false;
   } else if (process.platform === 'darwin' || process.platform === 'win32' || process.env.APPIMAGE) {
     ipcMain.on('autoUpdate', (event, args) => {
-      const disableUpdate = !params.settings.app.get('automaticUpdates');
 
-      if (disableUpdate) {
+      if (!enableUpdate) {
```
Comment (vantezzen) [inline src/electron/ipc-api/autoUpdate.js]: This should be `if (enableUpdate) {`

Comment (kytwb) [inline src/electron/ipc-api/autoUpdate.js]:
```suggestion
      if (enableUpdate) {
```

Comment (kytwb) [reply src/electron/ipc-api/autoUpdate.js]: I was trying code suggestions but I couldn't commit it directly via the UI in this case:

Screen Shot 2020-03-29 at 2 35 36 PM

Comment (kytwb) [reply src/electron/ipc-api/autoUpdate.js]: Addressed via https://github.com/getferdi/ferdi/pull/506/commits/53758fc53c4efe96c38d906be270f340370cf14f, thanks for the careful review, much appreciated 🙏

**Comment by yourcontact** [comment]
@vantezzen @kytwb I checked on my command line on the `refactor/settings-updates`. The branch was up-to-date:

Screen Shot 2020-04-01 at 4 08 51 PM

so I used "Resolved conflicts" on Github.
Let me know if that's what needed to be done.

**Comment by kytwb** [comment]
@yourcontact If you check the status of the last commit, you can see that the build failed:

Screen Shot 2020-04-02 at 8 44 40 AM

If you click [details](https://ci.appveyor.com/project/kytwb/ferdi/builds/31880291), you'll see the linter picked up some errors in your commit:

Screen Shot 2020-04-02 at 8 44 47 AM

**Comment by yourcontact** [comment]
@kytwb, it looks like the checks have passed. I am a bit confused though because the file 'recipes' still shows as _modified_. However, no changes show in my terminal when I run `git diff`.

Below is what I did:

- I ran `git fetch` to get the files from branch refactor/settings-updates
- I ran `npm run prepare-code`
- I fixed the error by deleting unused variable `server` https://github.com/getferdi/ferdi/pull/506/commits/f4cedf273dce9a89c2c93065a3f360824becbd88
- (Apply code style https://github.com/getferdi/ferdi/pull/506/commits/28ac271b54edbf1cdf9f1e5e33e2bd9bdf36fb4e)
- I ran `npm run prepare-code` again
- the file ‘recipes’  popped up as _modified_ when I ran git status
- No changes showed after I ran `git diff`
- Visual Studio Code still shows a 'M' (modified) next to the ‘recipes’

**Comment by vantezzen** [comment]
> changes show in my terminal when I run `git diff`

This is probably due to `recipes` being a submodule - the difference is that it uses a different commit on that repository now. You can commit those changes of submodules if you want to but normally we also update submodules on build so you won't have to do that.

**Comment by yourcontact** [comment]
@vantezzen, let me know if there is anything else to do on that PR.

**Comment by vantezzen** [comment]
The text for enabling updates still seems to be "Disable updates":

https://github.com/getferdi/ferdi/blob/314beafa090a323f53363980acad5c53e0727d03/src/containers/settings/EditSettingsScreen.js#L163

Also, what is the status on the settings migration?

**Comment by yourcontact** [comment]
> The default text for enabling updates still seems to be "Disable updates":

I updated this https://github.com/getferdi/ferdi/pull/506/commits/8a69271131136007f5265a5194bbbca157f65e0c

> I don't know about @kytwb but I think we should make the text "Enable updates" instead of the current "Enable automatic updates". The current one sounds like you can still use manual updates which (as far as I can see) isn't possible.

Makes sense. I updated this https://github.com/getferdi/ferdi/pull/506/commits/f793ef5598265ccd2809c00fe47cfef26a614a6b

> Also, what is the status on the settings migration?

@kytwb and I added settings migration a March 30th but I can't find the commit. Weird.
It currently shows in 'SettingsStore.js' locally: 

```
    if (!this.all.migration['5.4.4-beta.4-settings']) {
      this.actions.settings.update({
        type: 'app',
        data: {
          todoServer: 'isUsingCustomTodoService',
          customTodoServer: legacySettings.todoServer,
        },
      });

      this.actions.settings.update({
        type: 'migration',
        data: {
          '5.4.4-beta.4-settings': true,
        },
      });

      debug('Migrated old todo setting to new custom todo setting');
    }
```

**Comment by vantezzen** [comment]
Hmm yes, I saw that change some days ago but now it seems to be gone from the PR?

**Comment by yourcontact** [comment]
>> ~@kytwb and I added settings migration a March 30th but I can't find the commit. Weird.
It currently shows in 'SettingsStore.js' locally:~
>
> Hmm yes, I saw that change some days ago but now it seems to be gone from the PR?

This piece of code is the settings migration for #527  🤦‍♂️ .

Let me do settings migration for Settings Updates.

**Comment by yourcontact** [comment]
@vantezzen @kytwb, I'm tempted to do the migration settings of Updates like this:

    if (!this.all.migration['5.4.4-beta.4-settings']) {
      this.actions.settings.update({
        type: 'app',
        data: {
          automaticUpdates: legacySettings.noUpdates,
        },
      });

      this.actions.settings.update({
        type: 'migration',
        data: {
          '5.4.4-beta.4-settings': true,
        },
      });

      debug('Migrated updates settings');
    }``

however `automaticUpdates` is the inverse of `noUpdates`.

How can I migrate the "inverse" of Updates setting (if that makes sense)?

-------------

>>I don't know about @kytwb but I think we should make the text "Enable updates" instead of the current "Enable automatic updates". The current one sounds like you can still use manual updates which (as far as I can see) isn't possible.
>
>Makes sense. I updated this f793ef5

@vantezzen, since I updated the name of the text to "Enable updates", shall I update the name of `automaticUpdates` to `EnableUpdates`?

**Comment by kytwb** [comment]
@yourcontact The setting is still stored in `automaticUpdates`, so no need to rename anything for the migration.

> Since I updated the name of the text to "Enable updates", shall I update the name of automaticUpdates to EnableUpdates?

You are asking if you should rename the name of the variable the setting is stored in because you changed the text that is used as a label for that setting in the UI, which are two different things.  
We can clarify this in our next call if it's not clear 🙏

File: src/i18n/locales/defaultMessages.json [hunk 2]
```diff
@@ -4609,15 +4609,16 @@
...
         "end": {
           "column": 3,
           "line": 164
         },
         "file": "src/containers/settings/EditSettingsScreen.js",
-        "id": "settings.app.form.noUpdates",
+        "id": "settings.app.form.automaticUpdates",
         "start": {
-          "column": 13,
+
```
Comment (kytwb) [inline src/i18n/locales/defaultMessages.json:4620]: @yourcontact there seems to be an empty line here 🙏

Comment (yourcontact) [reply src/i18n/locales/defaultMessages.json]: Thank you for spotting this @kytwb I removed it. https://github.com/getferdi/ferdi/pull/506/commits/7a8a6cddcadbca1a053fef3c962d63b5048ab1af

**Comment by yourcontact** [comment]
>I think something got lost during one of the merge conflict resolution:
>Screenshot 2020-04-08 at 11 37 05
>Does it ring a bell? If not, please ignore it and I'll fix it on develop before the next release, it's a minor issue (or we can discuss it tomorrow during our pairing session).

I added fragments [as suggested](https://github.com/getferdi/ferdi/pull/506#issuecomment-610022931.). Then I checked the message "Show draggable area on window." It was as expected (without "!!!"), so I committed my changes.

I checked in Electron after the commit and this message got messed up:

Screenshot 2020-04-08 at 12 03 28

I went back to my terminal to check `git diff`. This piece of code was added after the commit:

Screenshot 2020-04-08 at 12 03 18

I looked for `en-US` file to fix "!!!Show draggable area on window" but I can't find any:

Screenshot 2020-04-08 at 12 14 54

Any suggestions on what went wrong?

**Comment by kytwb** [comment]
@yourcontact Can you then add it somewhere in `en-US.json` please? 🙏 The position in the file isn't important, but it would be better to add it near other strings with a similar `id` (`settings.app.form...`).

**Comment by yourcontact** [comment]
> @yourcontact Can you then add it somewhere in en-US.json please? 🙏 The position in the file isn't important, but it would be better to add it near other strings with a similar id (settings.app.form...).

I [added it](https://github.com/getferdi/ferdi/pull/506/commits/6495e28fd23730fc84aa5ecc4313b74b76ca7e97).

After commit https://github.com/getferdi/ferdi/pull/506/commits/6495e28fd23730fc84aa5ecc4313b74b76ca7e97 there were some code style changes applied. I also have these changes below.
Are those due to previous merges on `develop` branch?
What "commit message" do you use to commit those changes? 

Screenshot 2020-04-08 at 12 56 29

And when I scroll down a bit further I find this one:

Screenshot 2020-04-08 at 12 56 38

**Comment by kytwb** [comment]
You can use `Apply prepare-code script changes` as a commit message I think. The problem is that on your branch, prepare-code still run on the pre-push git hook. This has been addressed already via https://github.com/getferdi/ferdi/commit/2625ca0e34f7f5b35ef7fde86fdecb3b58f0f7e4 to run on the pre-commit hook instead to avoid this kind of confusions.

**Comment by kytwb** [comment]
@vantezzen I can't merge the pull request at the moment, not sure why:

Screen Shot 2020-04-09 at 1 01 13 AM

**Pull request merged**